### PR TITLE
First Task from Threads

### PR DESCRIPTION
**Pull request by p-nedelchev**
Needs a review

### CODE REVIEW

File: src/main/java/com/clouway/first/Demo.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * @author Petar Nedelchev <peter.krasimirov@gmail.com>
+ */
+public class Demo {
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        Counter counter = new Counter();
+        Thread countingThread = new Thread(counter);
+        while(true) {
```
Comment (tsony-tsonev) [inline src/main/java/com/clouway/first/Demo.java]: why we need this while

Comment (p-nedelchev) [inline src/main/java/com/clouway/first/Demo.java]: Actually it's not needed. It will wait for input anyway. I will fix it.

**Comment by p-nedelchev** [comment]
Fixes applied.

File: src/main/java/com/clouway/first/Counter.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                Thread.sleep(500);
+                counter++;
+            } catch (InterruptedException e) {
+                printValue();
+                break;
+            }
+        }
+    }
+
+    private void printValue() {
```
Comment (tsony-tsonev) [inline src/main/java/com/clouway/first/Counter.java]: why you replaced the getter method with this one

Comment (p-nedelchev) [inline src/main/java/com/clouway/first/Counter.java]: When interupt() method on the thread is called it throws an InterruptedException, thread is stopped and it prints the current value of the counter.

Comment (tsony-tsonev) [inline src/main/java/com/clouway/first/Counter.java]: I see that, but why did you preferred this approach. You can simply return the counter here as it was before, remove the logic from the catch block (bad practice) and in the main just print the counter value by its getter method after interrupting the thread.

Another note: don't interrupt threads, it is not a good solution, you can loop by some boolean flag in the run method and, then you can create another method responsible for stopping the thread by changing the `while` flag to false.

**Comment by p-nedelchev** [comment]
Changes made according to the comments. Now a flag controls the while cycle and there is a method that sets the flag to false and breaks the cycle. Getter is back and used to print counter value after thread is stopped.

**Comment by tsony-tsonev** [comment]
LGTM btw flag was example name

**Pull request merged**